codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -147,7 +130,7 @@ public void onHttpServerUpgrade(Http2Settings settings) throws Http2Exception {
     @Override
     public void channelActive(ChannelHandlerContext ctx) throws Exception {
         // The channel just became active - send the connection preface to the remote endpoint.
-        sendPreface(ctx);
+        init(ctx);
```

Comment:
@nmittler instead of calling `init()` here, could we do this as the last statement in the constructor? I know it's generally a anti-pattern to let `this` escape from within the constructor, but I think it's fine here? WDYT?